src/hidimstat/clustered_inference.py

```diff
@@ -65,11 +67,6 @@ def hd_inference(X, y, method, n_jobs=1, memory=None, verbose=0, **kwargs):
     n_jobs : int or None, optional (default=1)
         Number of CPUs to use during parallel steps such as inference.
 
-    memory : str or joblib.Memory object, optional (default=None)
```

Comment:
why do you remove the memory argument ?

Reply:
This argument is for optimisation of the calculation by memorising the results of a call of a function with the same arguments. I don't think the basic user requires it and I don't take the time to look in detail if it's very efficient.
I think that it's interesting when the function is run multiple times on the same data but I don't think that it's important to keep it for the moment because it should be the case.

Reply:
joblib caching is a pattern that is used quite systematically throughout sklearn and nilearn. I think it is useful in general for hidimstat because importance analysis typically requires recurrent computations.
So, if there is no obvious reason to remove it I suggest to keep it.

Reply:
What do you mean by recurrent computations?

Reply:
Cross validation and pipeline structure often lead to some computations being repeated with the same arguments. Using caching is extremely useful in such cases.

Reply:
In the use case that you identify, I don't see the usage of the recurrent computations.
If you define a specific use case where it's needed, I will modify the code accordingly.

Reply:
@bthirion ?

Reply:
I definitely want to keep  the caching.

Reply:
Which function do you want to catch?

Reply:
`desparsified_lasso`, as it is costly

Reply:
ok done

Reply:
Instead of  caching `desparsified_lasso`, I cache the function `hd_inference` which runs `desparsified_lasso` and runs the function for getting the pvalues.